pkg/services/ngalert/ngalert.go

```diff
@@ -171,7 +172,16 @@ func (ng *AlertNG) init() error {
 
 	decryptFn := ng.SecretsService.GetDecryptedValue
 	multiOrgMetrics := ng.Metrics.GetMultiOrgAlertmanagerMetrics()
-	ng.MultiOrgAlertmanager, err = notifier.NewMultiOrgAlertmanager(ng.Cfg, ng.store, ng.store, ng.KVStore, ng.store, decryptFn, multiOrgMetrics, ng.NotificationService, log.New("ngalert.multiorg.alertmanager"), ng.SecretsService)
+	multiOrgLogger := log.New("ngalert.multiorg.alertmanager")
+	if ng.Cfg.UnifiedAlerting.RemoteAlertmanager.Enable {
+		override := notifier.WithAlertmanagerOverride(func(ctx context.Context, orgID int64) (notifier.Alertmanager, error) {
+			externalAMCfg := remote.ExternalAlertmanagerConfig{}
+			return remote.NewExternalAlertmanager(externalAMCfg, orgID)
+		})
+		ng.MultiOrgAlertmanager, err = notifier.NewMultiOrgAlertmanager(ng.Cfg, ng.store, ng.store, ng.KVStore, ng.store, decryptFn, multiOrgMetrics, ng.NotificationService, multiOrgLogger, ng.SecretsService, override)
```

Comment:
It looks like the difference here is between `override` being nil or non-nil, so what about something like this to avoid the redundant calls to `NewMultiOrgAlertmanager`:

```go
var override Option
if ng.Cfg.UnifiedAlerting.RemoteAlertmanager.Enable {
		override = notifier.WithAlertmanagerOverride(func(ctx context.Context, orgID int64) (notifier.Alertmanager, error) {
			externalAMCfg := remote.ExternalAlertmanagerConfig{}
			return remote.NewExternalAlertmanager(externalAMCfg, orgID)
		})
}
ng.MultiOrgAlertmanager, err = notifier.NewMultiOrgAlertmanager(ng.Cfg, ng.store, ng.store, ng.KVStore, ng.store, decryptFn, multiOrgMetrics, ng.NotificationService, multiOrgLogger, ng.SecretsService, override)
```

Reply:
Thanks for the suggestion.